src/autotype/mac/AutoTypeMac.cpp

```diff
@@ -116,29 +110,34 @@ QString AutoTypePlatformMac::activeWindowTitle()
 //
 bool AutoTypePlatformMac::registerGlobalShortcut(Qt::Key key, Qt::KeyboardModifiers modifiers)
 {
-    uint16 nativeKeyCode = qtToNativeKeyCode(key);
+    CGKeyCode nativeKeyCode = qtToNativeKeyCode(key);
     if (nativeKeyCode == INVALID_KEYCODE) {
         qWarning("Invalid key code");
         return false;
     }
-    CGEventFlags nativeModifiers = qtToNativeModifiers(modifiers, false);
-    if (::RegisterEventHotKey(nativeKeyCode, nativeModifiers, m_hotkeyId, GetApplicationEventTarget(), 0, &m_hotkeyRef) != noErr) {
-        qWarning("Register hotkey failed");
-        return false;
-    }
-
+    CGEventFlags nativeModifiers = qtToNativeModifiers(modifiers);
+    m_globalMonitor =
+        macUtils()->addGlobalMonitor(nativeKeyCode, nativeModifiers, this, AutoTypePlatformMac::hotkeyHandler);
     return true;
 }
 
+//
+// Hotkey Handler
+//
+void AutoTypePlatformMac::hotkeyHandler(void* userData)
+{
+    AutoTypePlatformMac* self = static_cast<AutoTypePlatformMac*>(userData);
+    emit self->globalShortcutTriggered();
+}
+
 //
 // Unregister global hotkey
 //
 void AutoTypePlatformMac::unregisterGlobalShortcut(Qt::Key key, Qt::KeyboardModifiers modifiers)
 {
     Q_UNUSED(key);
     Q_UNUSED(modifiers);
-
-    ::UnregisterEventHotKey(m_hotkeyRef);
+    macUtils()->removeGlobalMonitor(m_globalMonitor);
```

Comment:
I noticed a problem with my implementation here. A segmentation fault occurs in `[NSEvent removeMonitor:]` if m_globalMonitor is non null and is passed more than once to removeMonitor.

Fix:
```
if (m_globalMonitor) {
    macUtils()->removeGlobalMonitor(m_globalMonitor);
    m_globalMonitor = nullptr;
}
```

Reply:
Yes I have just done this haha, found that same error